### PR TITLE
Enable Upstream Testing

### PR DESCRIPTION
**Pull request by bryanculver**
- Updated Dockerfile to be compatible with upstream testing
- Added workflow fragment to trigger testing

### CODE REVIEW

File: .github/workflows/upstream_testing.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+name: "Nautobot Upstream Testing"
+
+on: # yamllint disable-line rule:truthy rule:comments
+  schedule:
+    - cron: "0 4 */2 * *"  # every other day at midnight
+  push:  # yamllint disable rule:empty-values
+# TODO: REMOVE PUSH LINE ABOVE
```
Comment (bryanculver) [inline .github/workflows/upstream_testing.yml:8]:
```suggestion
```
Remove before merging.

File: development/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,73 @@
...
+#
+# We can't use the entire freeze as it takes forever to resolve with rigidly fixed non-direct dependencies,
+#   especially those that are only direct to Nautobot but the container included versions slightly mismatch
+RUN poetry export -f requirements.txt --without-hashes --output poetry_freeze_base.txt
+RUN poetry export -f requirements.txt --dev --without-hashes --output poetry_freeze_all.txt
+RUN sort poetry_freeze_base.txt poetry_freeze_all.txt | uniq -u > poetry_freeze_dev.txt
+
+# Install all local project as editable, constrained on Nautobot version, to get any additional
+#   direct dependencies of the plugin
+RUN pip install -c constraints.txt -e .
```
Comment (bryanculver) [inline development/Dockerfile:66]: This fails currently because this project isn't setup to be installed as editable.

**Comment by bryanculver** [comment]
`next` is expected to fail because it goes to a new major version of Nautobot so can be safely ignored for now.

**Comment by bryanculver** [comment]
Probably dependent on #29

**Comment by snaselj** [comment]
Rebased on top of `next`

File: .github/workflows/upstream.yml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 ---
-name: "Nautobot Upstream Testing"
+name: "Nautobot Upstream Monitor"
 
-on: # yamllint disable-line rule:truthy rule:comments
+on:  # yamllint disable-line rule:truthy rule:comments
   schedule:
     - cron: "0 4 */2 * *"  # every other day at midnight
-  push:  # yamllint disable rule:empty-values
+  push: {}
```
Comment (bryanculver) [inline .github/workflows/upstream.yml:7]: To be removed once before we merge to `develop`.

File: development/Dockerfile (hunk 2 above)
Comment (snaselj) [reply development/Dockerfile:71]: Fixed by: https://github.com/nautobot/nautobot-plugin-data-validation-engine/pull/30/commits/de31beb266f2119f2c6ef10bc2a623401e7fc5d9

**Comment by snaselj** [comment]
Closing as it's been fully implemented in #29

**Pull request closed without merging**